util/stream.go

```diff
@@ -126,14 +131,17 @@ func (m *MessageStream) outbound() {
 			}
 
 			klog.V(4).InfoS("Sent", "dataLength", len(data), "data", len(data), data)
+		case <-time.After(messageTimeout):
```

Comment:
Question: If we add timeout control in MessageStream outbound functions, do we need other similar control in ofnet?

Reply:
Maybe you meant this?

```
	for {
		select {
		case <-m.Shutdown:
			klog.Infof("Closing OpenFlow message stream.")
			m.conn.Close()
			close(m.parserShutdown)
			return
		case msg := <-m.Outbound:
			// Forward outbound messages to conn
			data, _ := msg.MarshalBinary()
			writeErrorCh := make(chan error)
			go func() {
				_, err := m.conn.Write(data)
				writeErrorCh<-err
			}()
			select {
			case err := <-writeErrorCh:
				if err != nil {
					klog.ErrorS(err, "OutboundError")
					m.Error <- err
					m.Shutdown <- true
				} else {
					klog.V(4).InfoS("Sent", "dataLength", len(data), "data", len(data), data)
				}
			case <-time.After(messageTimeout):
				m.Error <- errors.New("Write to socket timed out")
				m.Shutdown <- true
			}
			close(writeErrorCh)
		}
```

Reply:
My bad, put the timer under wrong scope. Would fix this.

Reply:
To your question:
Question: If we add timeout control in MessageStream outbound functions, do we need other similar control in ofnet?

No we should not have any timer in the ofnet code. ofnet code should either get a success or an error and should not expect libopenflow/stream code to block indefinitely.